cmd/keeper/keeper.go

```diff
@@ -842,6 +842,80 @@ func (p *PostgresKeeper) updateReplSlots(dbLocalState *DBLocalState, followersUI
 	return nil
 }
 
+func (p *PostgresKeeper) recreateAddonReplicationSlots(addonSlotNames []string) error {
+	var existingReplicationSlots []string
+	existingReplicationSlots, err := p.pgm.GetReplicationSlots()
+
+	if err != nil {
+		log.Errorw("failed to get replication slots", zap.Error(err))
+		return err
+	}
+
+	// detect not stolon replication slots
+	notStolonSlots := []string{}
+	for _, currReplicationSlotName := range existingReplicationSlots {
+		if !common.IsStolonName(currReplicationSlotName) {
+			notStolonSlots = append(notStolonSlots, currReplicationSlotName)
+		}
+	}
+
+	// detect slot that should be dropped
+	slotsToDrop := []string{}
+
+	for _, removeCandidateSlot := range notStolonSlots {
+		slotShouldSurvive := false
+		for _, replSlotName := range addonSlotNames {
+			if replSlotName == removeCandidateSlot {
+				slotShouldSurvive = true
+			}
+		}
+
+		if !slotShouldSurvive { // slot not detected as additional, so schedule the removal
+			slotsToDrop = append(slotsToDrop, removeCandidateSlot)
+		}
+	}
```

Comment:
This could be simplified in (not tested):

```
for _, rs := range notStolonSlots {
	if !util.StringInSlice(addonSlotNames, replSlot) {
		slotsToDrop = append(slotsToDrop, rs)
	}
}
```